### PR TITLE
ZWEB-12817: refactored, rewrote code from coffee script to js

### PR DESCRIPTION
**Pull request by gvozdetskyi-mykyta**
Jira Task: https://zimitio.atlassian.net/browse/ZWEB-12817

Steps to Reproduce:

- Run npm audit
- You should not notice any vulnerabilities
- Run npm run test
- Make sure that all tests are passed

Screenshot 2023-04-11 at 16 43 26

Screenshot 2023-04-11 at 16 43 47

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+const { addFiles, generateDocument } = require('./internal');
+const JSZip = require('jszip');
+
+module.exports = {
+    asBuffer,
+};
+
+async function asBuffer(html, options) {
+    const zip = new JSZip();
```
Comment (z-o-ilchenko) [inline lib/api.js:18]: According to this guide https://stuk.github.io/jszip/documentation/upgrade_guide.html `jszip` now works another way (without `new JSZip();`). Have you checked, is it working properly?

File: lib/internal.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const { readFile } = require('fs/promises');
+const { getMHTDocument } = require('./utils');
+const getDocumentTemplate = require('./templates/document');
+
+module.exports = {
+    generateDocument,
+    addFiles,
+};
+
+async function generateDocument(zip) {
```
Comment (z-o-ilchenko) [inline lib/internal.js:18]: Let's add docblocks with param types and descriptions to all functions (the same for utils).

**Review by z-o-ilchenko — CHANGES_REQUESTED** [review]

File: lib/internal.js (hunk 2 above)
Comment (gvozdetskyi-mykyta) [reply lib/internal.js:18]: Good idea

File: lib/api.js (hunk 1 above)
Comment (gvozdetskyi-mykyta) [reply lib/api.js:18]: Yes, I've checked. It works with `new JSZip();` https://stuk.github.io/jszip/documentation/examples.html. In that case you sent they use static method for uploading content, but in our case it's possible to use just an instance.

File: lib/internal.js (hunk 2 above)
Comment (gvozdetskyi-mykyta) [reply lib/internal.js:18]: done

**Review by z-o-ilchenko — APPROVED** [review]

**Review by ValeriiSyrov — APPROVED** [review]

**Pull request merged**